src/language-js/printer-estree.js

```diff
@@ -5083,26 +5082,44 @@ function printMemberChain(path, options, print) {
  * @param {import('estree').Node} node
  * @returns {boolean}
  */
-function isSimple(node) {
+function isSimple(node, depth = 0) {
+  if (depth >= 2) {
+    return false;
+  }
+  const isChildSimple = child => isSimple(child, depth + 1);
   if (
     !node ||
     node.type === "Literal" ||
     node.type === "BooleanLiteral" ||
     node.type === "NullLiteral" ||
     node.type === "NumericLiteral" ||
     node.type === "StringLiteral" ||
-    node.type === "Identifier"
+    node.type === "Identifier" ||
+    node.type === "OptionalCallExpression" ||
+    node.type === "OptionalMemberExpression" ||
+    node.type === "NewExpression" ||
```

Comment:
`OptionalCallExpression` and `NewExpression` are like `CallExpression`. `OptionalMemberExpression` is like `MemberExpression`. They should be moved to the corresponding conditions. `TSNonNullExpression` is like a unary expression, its operand needs to be checked too.